### ISSUE TITLE
Follow symlinks only once

### ISSUE THREAD

**Issue by mrzv** [sharkdp/fd#318]
It would be nice to add an option to follow a symlink only once. Or more accurately to visit a directory that a symlink points to only once. So if I have a directory A, and multiple symlinks point to it, it would be nice to have an option to report results from A only the first time it's encountered.

**Comment by sharkdp** [comment]
Thank you for the feedback!

This would probably be difficult to implement since it is not supported by the [ignore crate](https://crates.io/crates/ignore), as far as I can tell. It would probably require us to keep a buffer of all visited directories.

Here is a simple way to achieve the same result, using `fd`'s `--exec`/`-x` option and `readlink`:

```bash
▶ tree
.
├── a
│   └── foo
├── b -> a
└── c -> a

▶ fd -L foo
a/foo
b/foo
c/foo

▶ fd -L foo -x readlink -f | sort | uniq
/tmp/test/a/foo
```

**Comment by mrzv** [comment]
I see. Is there a way to do the same for files? What I'm really searching are all PDF files and I don't want duplicates, reached via different symlinks, reported. So the search is:
```
fd -e pdf -L
```

**Comment by sharkdp** [comment]
Yes, I believe the same command should work for symlinks to files(?).

**Comment by tmccombs** [comment]
would it make sense to add that functionality to the ignore crate? I imagine similar functionality would be useful for ripgrep as well.

**Comment by mrzv** [comment]
@sharkdp At least on a Mac, it doesn't work as intended. It will resolve a symlink, but not the input is a file contained in a symlink directory.

@tmccombs It would be great if this functionality was added to ripgrep as well (I'm a happy user of that, too). I was using fd to pipe filenames into pdfgrep, when I ran into this issue, but if I was after plain text, I would have run into this with ripgrep.

**Comment by tavianator** [comment]
I implemented this in bfs as the `-unique` option: https://github.com/tavianator/bfs/commit/b921fb2f43869ed12822ee99a80ccba9fd67c640

**Comment by sharkdp** [comment]
I'd like to close this. There are no plans to integrate this in `fd`. If this feature should appear in the `ignore` crate at some point, I'm happy to reconsider.
